tests/test_gameboard.py

```diff
@@ -146,6 +146,56 @@ def test_validate_matrix_with_invalid_matrix_wrong_structure(self):
             ['blue', 'blue',  'red', 'blue',  'red',   'blue',  None],
             ['red',   'red',  'red', 'blue',  'red',   'blue',  None]]))
 
+    def test_count_same_color_on_top_of_first_column(self):
```

Comment:
Is this the most simple example? I'm sure we discussed to start with the empty board, then with a board that has a single chip, then multiple chips with the same color and then reach this one.